File: Taskly Backend/src/main/java/com/Task_Management_System/TMS/service/TeamService.java
package com.Task_Management_System.TMS.service;

import com.Task_Management_System.TMS.dto.AddMemberDto;
import com.Task_Management_System.TMS.dto.TeamDto;
import com.Task_Management_System.TMS.dto.TeamMemberDto;
import com.Task_Management_System.TMS.dto.TeamResponseDto;

import java.util.List;
import java.util.UUID;

public interface TeamService {
    TeamResponseDto createTeam(TeamDto teamDto);
    List<TeamResponseDto> getAllTeams();
    TeamResponseDto getTeamById(UUID id);
    TeamResponseDto updateTeam(UUID id, TeamDto teamDto);
    void deleteTeam(UUID id);
    TeamMemberDto addMember(UUID teamId, AddMemberDto addMemberDto);
    void removeMember(UUID teamId, UUID userId);
    List<TeamMemberDto> getTeamMembers(UUID teamId);
}

File: Taskly Frontend/src/app/core/services/team.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Team, TeamMember } from '../models/team.model';

@Injectable({
  providedIn: 'root'
})
export class TeamService {
  private http = inject(HttpClient);
  private apiUrl = (window.location.origin.includes('localhost')
      ? 'http://localhost:8080/api'
      : '/api') + '/teams';

  createTeam(team: { name: string; ownerId: string }): Observable<Team> {
    return this.http.post<Team>(this.apiUrl, team);
  }

  getTeams(): Observable<Team[]> {
    return this.http.get<Team[]>(this.apiUrl);
  }

  getTeamById(id: string): Observable<Team> {
    return this.http.get<Team>(`${this.apiUrl}/${id}`);
  }

  addMember(teamId: string, member: { email: string; role: string }): Observable<TeamMember> {
    return this.http.post<TeamMember>(`${this.apiUrl}/${teamId}/members`, member);
  }

  getTeamMembers(teamId: string): Observable<TeamMember[]> {
    return this.http.get<TeamMember[]>(`${this.apiUrl}/${teamId}/members`);
  }

  removeMember(teamId: string, userId: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${teamId}/members/${userId}`);
  }

  deleteTeam(teamId: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${teamId}`);
  }
}
